### PR TITLE
Fixes from mount move testing

### PR DESCRIPTION
**Pull request by pmmukh**
- Adds a check for spaces in the source or destination mounts
- Updates the auth move docs
- Besides those 2 fixes from the manual testing, the PR also adds the changelog for the feature

Also, thanks to @akshya96  for all the testing catches!

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -1201,6 +1201,13 @@ func (b *SystemBackend) handleRemount(ctx context.Context, req *logical.Request,
 			logical.ErrInvalidRequest
 	}
 
+	if strings.Contains(fromPath, " ") {
+		return logical.ErrorResponse("'from' path cannot contain whitespaces"), logical.ErrInvalidRequest
```
Comment (raskchanky) [inline vault/logical_system.go:1205]:
```suggestion
		return logical.ErrorResponse("'from' path cannot contain whitespace"), logical.ErrInvalidRequest
```

File: vault/logical_system.go [hunk 2]
```diff
@@ -1201,6 +1201,13 @@ func (b *SystemBackend) handleRemount(ctx context.Context, req *logical.Request,
...
+	}
+	if strings.Contains(toPath, " ") {
+		return logical.ErrorResponse("'to' path cannot contain whitespaces"), logical.ErrInvalidRequest
```
Comment (raskchanky) [inline vault/logical_system.go:1208]:
```suggestion
		return logical.ErrorResponse("'to' path cannot contain whitespace"), logical.ErrInvalidRequest
```

File: vault/logical_system_test.go [hunk 3]
```diff
@@ -995,6 +995,38 @@ func TestSystemBackend_remount_nonPrintable(t *testing.T) {
...
+
+	req := logical.TestRequest(t, logical.UpdateOperation, "remount")
+	req.Data["from"] = " foo / "
+	req.Data["to"] = "bar"
+	req.Data["config"] = structs.Map(MountConfig{})
+	resp, err := b.HandleRequest(namespace.RootContext(nil), req)
+	if err != logical.ErrInvalidRequest {
+		t.Fatalf("err: %v", err)
+	}
+	if resp.Data["error"] != `'from' path cannot contain whitespaces` {
```
Comment (raskchanky) [inline vault/logical_system_test.go:1009]:
```suggestion
	if resp.Data["error"] != `'from' path cannot contain whitespace` {
```

File: vault/logical_system_test.go [hunk 4]
```diff
@@ -995,6 +995,38 @@ func TestSystemBackend_remount_nonPrintable(t *testing.T) {
...
+
+	req := logical.TestRequest(t, logical.UpdateOperation, "remount")
+	req.Data["from"] = "foo"
+	req.Data["to"] = " bar / "
+	req.Data["config"] = structs.Map(MountConfig{})
+	resp, err := b.HandleRequest(namespace.RootContext(nil), req)
+	if err != logical.ErrInvalidRequest {
+		t.Fatalf("err: %v", err)
+	}
+	if resp.Data["error"] != `'to' path cannot contain whitespaces` {
```
Comment (raskchanky) [inline vault/logical_system_test.go:1025]:
```suggestion
	if resp.Data["error"] != `'to' path cannot contain whitespace` {
```

**Review by raskchanky — APPROVED** [review]
A few nits, looks good!

**Review by akshya96 — APPROVED** [review]

**Pull request merged**